File: frontend/src/services/paymentService.ts
import { paymentApi } from "@/apis/paymentApi";
import type {
  PaymentReadyInfo,
  PaymentVerifyRequest,
  PaymentVerifyResult,
} from "@/types/payment";

/**
 * 결제 도메인 유스케이스를 표현하는 서비스 계층.
 * UI는 이 계층만 호출하고, 실제 HTTP는 api 계층에 위임한다.
 */
export const paymentService = {
  prepareTestPayment(signal?: AbortSignal): Promise<PaymentReadyInfo> {
    return paymentApi.ready(signal);
  },

  verifyPayment(
    request: PaymentVerifyRequest,
    signal?: AbortSignal
  ): Promise<PaymentVerifyResult> {
    return paymentApi.verify(request, signal);
  },
};

File: backend/src/main/java/com/spharos/payment/service/PaymentService.java
package com.spharos.payment.service;

import com.spharos.payment.dto.request.BillingPaymentRequest;
import com.spharos.payment.dto.request.PaymentVerifyRequest;
import com.spharos.payment.dto.response.PaymentReadyResponse;
import com.spharos.payment.dto.response.PaymentVerifyResponse;

public interface PaymentService {

    PaymentReadyResponse ready();

    PaymentVerifyResponse verify(PaymentVerifyRequest request);

    PaymentVerifyResponse payWithBillingKey(BillingPaymentRequest request);
}
